slippy/tile_grid.go

```diff
@@ -0,0 +1,107 @@
+package slippy
+
+import (
+	"fmt"
+	"math"
+)
+
+// TileGrid contains the tile layout, including ability to get WGS84 coordinates for tile extents
+type TileGrid interface {
+	ContainsIndex(zoom, x, y uint) bool
+	GridSize(zoom uint) (x, y uint)
+	MaxXY(zoom uint) (maxx, maxy uint)
+	Lat2YIndex(zoom uint, lat float64) (gridy uint)
+	Lon2XIndex(zoom uint, lon float64) (gridx uint)
+	XIndex2Lon(zoom, x uint) (lon float64)
+	YIndex2Lat(zoom, y uint) (lat float64)
+}
+
+func GetGrid(srid uint) TileGrid {
```

Comment:
For convention, this function should be renamed `NewGrid`. Also, as a clarification, are there external  limitations (other programs, a spec, convention etc.) which prevent the usage of square tiles with 4326 projection?

Reply:
I believe that, from my experience, when someone mentions "4326" as the projection (which it is not), they are typically referring to equirectangular projection using lat/lon and WGS84 geoid upon a 2x1 tile ratio (using square tiles, but 2x1 tile extent).  But that is completely notional, not part of a spec and my experience is only a single data-point in usages.

Reply:
I think I understand what's going on bit better now, thank you for the explanation. I always get confused with srid, projections, and datums. From what I've found after a brief search it seem like some programs expect the behavior you describe. So, I think the functionality here is good.

Reply:
I hear ya, I still get confused as to difference of them as well.